controllers/volumereplication_controller.go

```diff
@@ -317,6 +325,64 @@ func (r *VolumeReplicationReconciler) SetupWithManager(mgr ctrl.Manager, cfg *co
 		WithEventFilter(pred).Complete(r)
 }
 
+func (r *VolumeReplicationReconciler) waitForCrds() error {
+
+	logger := r.Log.WithName("Wait for VolumeReplicationClass and VolumeReplication crds")
+
+	err := r.waitForVolumeReplicationClass(logger)
+	if err != nil {
+		logger.Error(err, "Failed to watch VolumeReplicationClass")
+		return err
+	}
+
+	err = r.waitForVolumeReplication(logger)
+	if err != nil {
+		logger.Error(err, "Failed to watch VolumeReplication")
+		return err
+	}
+	return nil
+}
+
+func (r *VolumeReplicationReconciler) waitForVolumeReplicationClass(logger logr.Logger) error {
+
+	vrcObj := &replicationv1alpha1.VolumeReplicationClass{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Name: ""}, vrcObj)
```

Comment:
Instead of Get (which is for a specific resource) should we just attempt a List here? (@Madhu-1  thoughts?)

It seems odd that we expect errors that states the object is invalid (or unknown), than Name is empty initially, which a List can help disambiguate.

(applies to `waitForVolumeReplication` as well)

Reply:
done